lib/nostrum/cache/guild/guild_server.ex

```diff
@@ -339,20 +339,21 @@ defmodule Nostrum.Cache.Guild.GuildServer do
     {:reply, tf.(state), state}
   end
 
-  def handle_call({:update, guild}, _from, state) do
+  def handle_call({:update, guild}, _from, %Guild{} = state) do
+    guild = guild |> Guild.to_struct()
     {:reply, {state, guild}, guild}
   end
 
-  def handle_call({:delete}, _from, state) do
+  def handle_call({:delete}, _from, %Guild{} = state) do
     {:stop, :normal, state, %{}}
   end
 
-  def handle_call({:create, :member, guild_id, member}, _from, state) do
-    new_members = [member | state.members]
+  def handle_call({:create, :member, guild_id, member}, _from, %Guild{} = state) do
+    new_members = [member |> Guild.Member.to_struct() | state.members]
     {:reply, {guild_id, member}, %{state | members: new_members}}
```

Comment:
Another issue I can see here (and it's replicated throughout the module) is that the info we're sending will not be in a struct. Here, the state of the guild will have this member as a struct, but the event that the user handles will be a map (when according to documentation it should be a struct).

Do you think it would be better to move the conversion logic to the client side of the server? We could also move it back to the dispatch module, although it may be weird there as the other caches are stored-ish in ETS and as such won't need this type of conversion. Of course we could also just do the conversion first on the server side. 

Let me know your thoughts on this!